sphinx/ext/autodoc/__init__.py

```diff
@@ -530,6 +530,11 @@ def add_directive_header(self, sig: str) -> None:
             # Be explicit about the module, this is necessary since .. class::
             # etc. don't support a prepended module name
             self.add_line('   :module: %s' % self.modname, sourcename)
+        if self.real_modname != self.modname:
+            # Where the object is actually defined when there is a difference
+            # between the imported modname and the real modname.
+            self.add_line('   :canonical: %s.%s' % (self.real_modname, self.object_name),
```

Comment:
This does not support `import ... as ...` style imports. We need to obtain the real name of the subject

Reply:
Thanks for this.  This line was already different in the upcoming commit to behave properly for non top-level objects (like classes within a class).

I'll be really interested in your feedback when this is pushed.